tgui/packages/tgui/interfaces/StripMenu.tsx

```diff
@@ -0,0 +1,410 @@
+    displayName: "gloves",
+    gridSpot: getGridSpotKey([2, 3]),
+    image: "inventory-gloves.png",
+  },
+
+  right_hand: {
+    displayName: "right hand",
+    gridSpot: getGridSpotKey([2, 4]),
+    image: "inventory-hand_r.png",
+    additionalComponent: <CornerText align="left">R</CornerText>,
+  },
+
+  left_hand: {
+    displayName: "left hand",
+    gridSpot: getGridSpotKey([2, 5]),
+    image: "inventory-hand_l.png",
+    additionalComponent: <CornerText align="right">L</CornerText>,
+  },
+
+  shoes: {
+    displayName: "shoes",
+    gridSpot: getGridSpotKey([3, 2]),
+    image: "inventory-shoes.png",
+  },
+
+  suit_storage: {
+    displayName: "suit storage item",
+    gridSpot: getGridSpotKey([4, 0]),
+    image: "inventory-suit_storage.png",
+  },
+
+  id: {
+    displayName: "ID",
+    gridSpot: getGridSpotKey([4, 1]),
+    image: "inventory-id.png",
+  },
+
+  belt: {
+    displayName: "belt",
+    gridSpot: getGridSpotKey([4, 2]),
+    image: "inventory-belt.png",
+  },
+
+  back: {
+    displayName: "backpack",
+    gridSpot: getGridSpotKey([4, 3]),
+    image: "inventory-back.png",
+  },
+
+  left_pocket: {
+    displayName: "left pocket",
+    gridSpot: getGridSpotKey([4, 4]),
+    image: "inventory-pocket.png",
+  },
+
+  right_pocket: {
+    displayName: "right pocket",
+    gridSpot: getGridSpotKey([4, 5]),
+    image: "inventory-pocket.png",
+  },
+};
+
+enum ObscuringLevel {
+  Completely = 1,
+  Hidden = 2,
+}
+
+type Interactable = {
+  interacting: BooleanLike;
+};
+
+/**
+ * Some possible options:
+ *
+ * null - No interactions, no item, but is an available slot
+ * { interacting: 1 } - No item, but we're interacting with it
+ * { icon: icon, name: name } - An item with no alternate actions
+ *   that we're not interacting with.
+ * { icon, name, interacting: 1 } - An item with no alternate actions
+ *   that we're interacting with.
+ */
+type StripMenuItem =
+  | null
+  | Interactable
+  | ((
+      | {
+          icon: string;
+          name: string;
+          alternate?: string;
+        }
+      | {
+          obscured: ObscuringLevel;
+        }
+    ) &
+      Partial<Interactable>);
+
+type StripMenuData = {
+  items: Record<keyof typeof SLOTS, StripMenuItem>;
+  name: string;
+};
+
+export const StripMenu = (props, context) => {
+  const { act, data } = useBackend<StripMenuData>(context);
+
+  const gridSpots = new Map<GridSpotKey, string>();
+  for (const key of Object.keys(data.items)) {
+    gridSpots.set(SLOTS[key].gridSpot, key);
+  }
+
+  const grid = [];
+
+  for (let rowIndex = 0; rowIndex < ROWS; rowIndex++) {
+    const buttons = [];
+
+    for (let columnIndex = 0; columnIndex < COLUMNS; columnIndex++) {
+      const keyAtSpot = gridSpots.get(getGridSpotKey([rowIndex, columnIndex]));
+      if (!keyAtSpot) {
+        buttons.push(
+          <Stack.Item
+            style={{
+              width: BUTTON_DIMENSIONS,
+              height: BUTTON_DIMENSIONS,
+            }}
+          />
```

Comment:
These things are missing the keys in all push calls, and ESLint doesn't catch this pattern.

Honestly, I advise against this imperative approach, I think you can totally do it declaratively.

Reply:
Interested to see how this'd be done less imperatively, how might that look like?

Reply:
Yeah, I just mean map (over a range of numbers).